frontend/packages/console-app/src/components/tour/tour-context.ts

```diff
@@ -0,0 +1,86 @@
+import { createContext, useReducer, Reducer, Dispatch, ReducerAction } from 'react';
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore: FIXME missing exports due to out-of-sync @types/react-redux version
+import { useSelector } from 'react-redux';
+import { getFlagsObject } from '@console/internal/reducers/features';
+import { useExtensions } from '@console/plugin-sdk';
+import { RootState } from '@console/internal/redux';
+import { isGuidedTour, GuidedTour } from '@console/plugin-sdk/src/typings/guided-tour';
+import { TourActions } from './const';
+import { getTourLocalStorageForPerspective, filterTourBasedonPermissionAndFlag } from './utils';
+import { TourDataType } from './type';
+
+export const tourReducer = (state: TourState, action: TourActions) => {
+  const { stepNumber } = state;
+  switch (action) {
+    case TourActions.start:
+      return { startTour: true, closeTour: false, completedTour: false, stepNumber: 0 };
+    case TourActions.next:
+      return {
+        startTour: false,
+        closeTour: false,
+        completedTour: false,
+        stepNumber: stepNumber + 1,
```

Comment:
Better to use object spread operator to create a new state like `{ ...state, ...newState }`. It makes it clear what exactly are you updating for this specific action. Same comment for all the actions.
```suggestion
        ...state,
        stepNumber: stepNumber + 1,
```